src/redis_index.c

```diff
@@ -273,6 +273,38 @@ int Redis_LoadDocument(RedisSearchCtx *ctx, RedisModuleString *key, Document *do
   return REDISMODULE_OK;
 }
 
+int Redis_LoadDocumentEx(RedisSearchCtx *ctx, RedisModuleString *key, RedisModuleString **fields,
+                         size_t nfields, Document *doc) {
+  if (!fields) {
+    return Redis_LoadDocument(ctx, key, doc);
+  }
+
+  // Get the key itself
+  RedisModuleKey *rkey = RedisModule_OpenKey(ctx->redisCtx, key, REDISMODULE_READ);
+  if (rkey == NULL) {
+    return REDISMODULE_ERR;
+  }
+
+  if (RedisModule_KeyType(rkey) != REDISMODULE_KEYTYPE_HASH) {
+    RedisModule_CloseKey(rkey);
+    return REDISMODULE_ERR;
+  }
+
+  doc->fields = malloc(sizeof(*doc->fields) * nfields);
+
+  for (size_t ii = 0; ii < nfields; ++ii) {
+    int rv =
+        RedisModule_HashGet(rkey, REDISMODULE_HASH_NONE, fields[ii], &doc->fields[ii].text, NULL);
+    if (rv == REDISMODULE_OK) {
+      doc->numFields++;
+      doc->fields[ii].name = fields[ii];
+    }
+  }
+
+  RedisModule_CloseKey(rkey);
```

Comment:
Yeah, but we're dealing with a lot of keys here, it's better if we close them incrementally rather than end up with having thousands of opened keys at once. No?